platform/legacy/LegacyPlatformConfig.ts

```diff
@@ -0,0 +1,119 @@
+import { RawConfig } from '../config/RawConfigService';
+import { ConfigPath } from '../config/ConfigService';
+
+/**
+ * Represents legacy Kibana config class.
+ * @internal
+ */
+export interface LegacyConfig {
+  get: (configPath: string) => any;
+  set: (configPath: string, configValue: any) => void;
+  has: (configPath: string) => boolean;
+}
+
+/**
+ * Represents logging config supported by the legacy platform.
+ */
+interface LegacyLoggingConfig {
+  silent: boolean;
+  verbose: boolean;
+  quiet: boolean;
+  dest: string;
+  json: boolean;
+}
+
+/**
+ * Represents adapter between config provided by legacy platform and `RawConfig`
+ * supported by the current platform.
+ */
+export class LegacyConfigToRawConfigAdapter implements RawConfig {
+  constructor(private readonly legacyConfig: LegacyConfig) {}
+
+  get(configPath: ConfigPath) {
+    configPath = LegacyConfigToRawConfigAdapter.flattenConfigPath(configPath);
+
+    // Explicitly disable plugins until they are supported by the legacy config.
+    if (!this.legacyConfig.has(configPath) && configPath.endsWith('.enabled')) {
```

Comment:
Not sure what would be the best solution here. New platform checks  `enabled` status for every plugin (eg. `pid.enabled`), but old platform doesn't know these keys and throws error. So special casing and disabling such plugins was the simplest solution.